### PR TITLE
feat: improved Builder design

### PR DESCRIPTION
**Pull request by Larsklopstra**
This is the improved design for the builder featuring:

- Visually better grouping
- More breathing space
- Show "Add block" button on mouseover
- Improved dropdown contrast
- Show iteration count in header (useful for scanning blocks)

image

### CODE REVIEW

File: packages/forms/resources/views/components/builder.blade.php [hunk 1]
```diff
@@ -126,18 +131,12 @@ class="flex items-center justify-center h-8 w-8 rounded-full border border-gray-
 
         @if ((! $isItemCreationDisabled()) && (blank($getMaxItems()) || ($getMaxItems() > $getItemsCount())))
             <div x-data="{ isCreateButtonDropdownOpen: false }" class="relative flex justify-center">
-                <button
+                <x-filament::button
```
Comment (danharrin) [inline packages/forms/resources/views/components/builder.blade.php:134]: We cannot use `filament::button` within the forms package, however a `forms::button` button component is being introduced in #2057

**Comment by Larsklopstra** [comment]
> Can you post a dark mode screenshot too please? :)

The well uses a darker gray so the cards and inputs remain the same as other components

image

File: packages/forms/resources/views/components/builder.blade.php (hunk 1 above)
Comment (Larsklopstra) [reply packages/forms/resources/views/components/builder.blade.php:134]: Will this be merged soon? Then I'll refactor it already

**Comment by Larsklopstra** [comment]
> Can we make the same changes to the repeater?

Yes!

**Comment by ryangjchandler** [comment]
I wonder if we should replicate the border around the block actions on the right hand side too (delete button), perhaps a `border-l`?

**Comment by Larsklopstra** [comment]
> I wonder if we should replicate the border around the block actions on the right hand side too (delete button), perhaps a `border-l`?

Would look too "blocky", there's no need for a divider on that side because the whitespace already makes for a great divider :)

File: packages/forms/resources/views/components/builder.blade.php (hunk 1 above)
Comment (danharrin) [reply packages/forms/resources/views/components/builder.blade.php:134]: This will be merged first

**Comment by Larsklopstra** [comment]
@danharrin added the opt-in containment that will add the well, by default it has none

**Comment by johncarter-** [comment]
Any chance of a collapse/expand on section headers, with an option to toggle expand collapse on all at the top?

**Comment by ryangjchandler** [comment]
> Any chance of a collapse/expand on section headers, with an option to toggle expand collapse on all at the top?

Yeah, I like this idea. Being able to collapse a block, etc.

**Comment by Larsklopstra** [comment]
> > Any chance of a collapse/expand on section headers, with an option to toggle expand collapse on all at the top?
> 
> Yeah, I like this idea. Being able to collapse a block, etc.

Sounds good, maybe for a later PR? Let's get this merged first 👀

**Comment by danharrin** [comment]
We already have the x-collapse plugin installed, so it shouldn't be too time consuming to do it now @Larsklopstra?

**Comment by danharrin** [comment]
Todo - merge in #2261

**Comment by Larsklopstra** [comment]
Added expand (all) & collapse (all)

image

File: packages/forms/src/Components/Builder.php [hunk 2]
```diff
@@ -225,6 +227,13 @@ public function disableItemDeletion(bool | Closure $condition = true): static
         return $this;
     }
 
+    public function contained(bool | Closure $condition = true): static
```
Comment (danharrin) [inline packages/forms/src/Components/Builder.php:230]: Maybe we should call this feature "inset"? "Contained" sounds not clear enough

Comment (Larsklopstra) [reply packages/forms/src/Components/Builder.php]: Sounds good

**Comment by danharrin** [comment]
Todo: merge in #2057 and utilize `<x-forms::button>`

**Comment by danharrin** [comment]
Todo: reflex changes on repeater

**Comment by danharrin** [comment]
Todo: utilitize the dropdown.item blade component

**Comment by danharrin** [comment]
Merged #2057 into this, when that is merged this will be too.

**Pull request merged**